src/server.rs

```diff
@@ -191,10 +191,14 @@ struct CommandsHandle {
 impl Drop for CommandsHandle {
     fn drop(&mut self) {
         debug!("dropping commands {:?}", self.commands);
-        if let Err(err) = self.cleanup.send(CleanupRequest::Commands(
-            self.commands.clone(),
-        )) {
-            warn!("failed to notify cleanup of commands {:?}: {}", self.commands, err);
+        if let Err(err) = self
+            .cleanup
+            .send(CleanupRequest::Commands(self.commands.clone()))
```

Comment:
I'm not familiar with `Drop` semantics here. Do we need to manually clean up `self.commands` as well, or is it fine to just `clone()` and the vec will get cleaned later?

Reply:
I'm actually a little sad about this - theoretically we should just be able to move self.commands out of self (it is being dropped after all) but we have to clone it because we don't have quite that much control. Everything should properly be cleaned up without any additional work.